### PR TITLE
[DSSv3] Fix Issue #21100: Add data() function to std.container.array.Array

### PR DESCRIPTION
**Pull request by dandrei279**
Add a function called `data()` to return the internal representation of the Array.
The function is marked as `@system`

### CODE REVIEW

**Comment by maxhaton** [comment]
What does "DSSv3" refer to? 

Also see buildkite CI logs, style check failed.

File: std/container/array.d [hunk 1]
```diff
@@ -544,6 +544,17 @@ if (!is(immutable T == immutable bool))
         return _data.refCountedStore.isInitialized ? _data._capacity : 0;
     }
 
+    /**  
+     * Returns: the internal representation of the array.
+     *
+     * Complexity: $(BIGOH 1).
+     */
+
+    T[] data() @system
```
Comment (maxhaton) [inline std/container/array.d:553]: Unit test? Preferably one included in the documentation a la https://dlang.org/spec/unittest.html#documented-unittests

**Comment by dandrei279** [comment]
> What does "DSSv3" refer to?

DSSv3 stands for "D Summer School v3". We use this tag to make it easier for the organizers to see what PRs were made during the course.

**Comment by maxhaton** [comment]
> > What does "DSSv3" refer to?
> 
> DSSv3 stands for "D Summer School v3". We use this tag to make it easier for the organizers to see what PRs were made during the course.

OK Cool, for future reference if you mention the issue number in one of the commit messages our bot will pick it up and automate bugzilla-admin for you.

**Review by maxhaton — APPROVED** [review]

**Comment by MoonlightSentinel** [comment]
There were two major flaws with this implementation:

- crashes for `Array.init`: #8310 
- doesn't work with `const` / `immutable`: #8311

**Pull request merged**